File: ReinterpretAlter.py
# This Python file uses the following encoding: utf-8

from ClassStorer import ClassStorer


class ReinterpretAlter:
    def __init__(self):
        pass


    def removeReinterpret(self, classList):
        print("Removing interpret statements...")

        for cls in classList:
            lines = []
            print(cls, "removing interprets...", end="", flush=True)
            with open(ClassStorer.export_dir + "/" + cls + ".cpp") as fr:
                for line in fr:
                    indy = line.find("reinterpret_cast<")
                    while indy >= 0:
                        #clammer = 1 # include removed clammer
                        #for c in line[0:indy]:
                        #    if c == "(":
                        #        clammer += 1
                        tempx = len(line[indy:].split("(")[0]) + 1 #2 # clammer plus exclusive end
                        line = line[0:indy] + line[tempx:] # remove all in between
                        #closeClammer = 0
                        #checls = -1
                        #iff = len(line) - 1
                        #for c in line[iff:0:-1]:
                        #    if c == ")":
                        #        closeClammer += 1
                        #    if closeClammer == clammer:
                        #        checls = iff
                        #        break
                        #    iff -= 1
                        #line = line[0:checls] + line[checls + 1:] # remove closing clammer
                    lines.append(line)

            with open(ClassStorer.export_dir + "/" + cls + ".cpp", "w") as fw:
                for line in lines:
                    fw.write(line + "\n")

            print("DONE")

File: cpp_proj/reinterpretalter.cpp
#include "reinterpretalter.h"
#include "classstorer.h"


void ReinterpretAlter::removeReinterpret(vector<RawClass> classList)
{
  cout << "Removing interpret statements..." << endl;

  foreach (RawClass cls, classList) {
      QStringList lines;

#if DEBUGMODE
      cout << cls.getName().toStdString().c_str() << " removing interprets...";
#endif

      QFile file(ClassStorer::ExportDir + QString("/") + cls.getName() + QString(".cpp"));
      if(!file.open(QIODevice::ReadOnly)) {
          cout << "REINTERPRETALTER 1 ERROR: File could not be opened - " << cls.getName().toStdString().c_str() << ".cpp" << endl;
      }

      QTextStream in(&file);
      while(!in.atEnd()) {
          QString line = in.readLine().remove(QRegExp("([\n]{1})$"));
          int indy = line.indexOf("reinterpret_cast<");
          while (indy >= 0) {
              int tempx = line.right(line.length() - indy).split('(')[0].length(); // clammer plus exclusive end
              line = line.left(indy) + line.right(line.length() - indy - tempx); // remove all in between
              indy = line.indexOf("reinterpret_cast<");
          }
          lines.append(line);
      }
      file.close();


      QFile file2(ClassStorer::ExportDir + QString("/") + cls.getName() + QString(".cpp"));
      if(!file2.open(QIODevice::WriteOnly)) {
          cout << "REINTERPRETALTER 2 ERROR: File could not be opened - " << cls.getName().toStdString().c_str() << ".cpp" << endl;
      }

      QTextStream out(&file2);
      foreach (QString line, lines) {
          out << line.toStdString().c_str() << Qt::endl;
      }
      file2.close();

#if DEBUGMODE
      cout << "DONE" << endl;
#endif
  }
}
